src/Tezos/Forging.cpp

```diff
@@ -72,68 +72,92 @@ Data forgeZarith(uint64_t input) {
-
-  if (operation.kind() == Operation_OperationKind_REVEAL) {
-      auto publicKey = PublicKey(operation.reveal_operation_data().public_key(), TWPublicKeyTypeED25519);
-      forged.push_back(0x07);
-      append(forged, forgedSource);
-      append(forged, forgedFee);
-      append(forged, forgedCounter);
-      append(forged, forgedGasLimit);
-      append(forged, forgedStorageLimit);
-      append(forged, forgePublicKey(publicKey));
-      return forged;
-  } else if (operation.kind() == Operation_OperationKind_ORIGINATION) {
-      auto managerPublicKey = operation.origination_operation_data().manager_pubkey();
-      auto balance = operation.origination_operation_data().balance();
-      forged.push_back(0x09);
-      append(forged, forgedSource);
-      append(forged, forgedFee);
-      append(forged, forgedCounter);
-      append(forged, forgedGasLimit);
-      append(forged, forgedStorageLimit);
-      append(forged, forgePublicKeyHash(managerPublicKey));
-      append(forged, forgeZarith(balance));
-      append(forged, forgeBool(true));
-      append(forged, forgeBool(true));
-      append(forged, forgeBool(false));
-      append(forged, forgeBool(false));
-      return forged;
-  } else if (operation.kind() == Operation_OperationKind_DELEGATION) {
-      auto delegate = operation.delegation_operation_data().delegate();
-      forged.push_back(0x0a);
-      append(forged, forgedSource);
-      append(forged, forgedFee);
-      append(forged, forgedCounter);
-      append(forged, forgedGasLimit);
-      append(forged, forgedStorageLimit);
-      if (!delegate.empty()) {
-          append(forged, forgeBool(true));
-          append(forged, forgePublicKeyHash(delegate));
-      } else {
-          append(forged, forgeBool(false));
-      }
-      return forged;
-  } else {
-      auto forgedAmount = forgeZarith(operation.transaction_operation_data().amount());
-      auto forgedDestination = Address(operation.transaction_operation_data().destination()).forge();
-      forged.push_back(0x08);
-      append(forged, forgedSource);
-      append(forged, forgedFee);
-      append(forged, forgedCounter);
-      append(forged, forgedGasLimit);
-      append(forged, forgedStorageLimit);
-      append(forged, forgedAmount);
-      append(forged, forgedDestination);
-      append(forged, forgeBool(false));
-      return forged;
-  }
-  return Data();
+    auto forged = Data();
+    auto source = Address(operation.source());
+    auto forgedSource = source.forge();
+    auto forgedFee = forgeZarith(operation.fee());
+    auto forgedCounter = forgeZarith(operation.counter());
+    auto forgedGasLimit = forgeZarith(operation.gas_limit());
+    auto forgedStorageLimit = forgeZarith(operation.storage_limit());
+
+    if (operation.kind() == Operation_OperationKind_REVEAL) {
+        auto publicKey = PublicKey(operation.reveal_operation_data().public_key(), TWPublicKeyTypeED25519);
+        auto forgedPublicKey = forgePublicKey(publicKey);
```

Comment:
small nit: I find this easier to read if you append to the end. so things happen in order.